internal/internal_decision_state_machine.go

```diff
@@ -388,7 +420,7 @@ func (d *commandStateMachineBase) handleInitiatedEvent() {
 	switch d.state {
 	case commandStateCommandSent:
 		d.moveState(commandStateInitiated, eventInitiated)
-	case commandStateCanceledBeforeInitiated:
+	case commandStateCanceledBeforeInitiated, commandStateCanceledBeforeSent, commandStateCancellationCommandSent:
```

Comment:
This is super confusing.  There are many state machines where we make requestCancel first and then it is accepted or rejected.  Like activity and child workflow cancellation.
I don't think following state transition is valid:
commandStateCancellationCommandSent -> commandStateCanceledAfterInitiated

It should be `commandStateRequestCancel`.

Reply:
It certainly is confusing, heh - to be clear you're suggesting adding a new state `commandStateRequestCancel` and using that rather than `commandStateCanceledAfterInitiated` here?